flink-scala-shell/src/main/java/org/apache/flink/api/java/ScalaShellRemoteStreamEnvironment.java

```diff
@@ -68,30 +67,21 @@ public ScalaShellRemoteStreamEnvironment(
 		this.flinkILoop = flinkILoop;
 	}
 
-	/**
-	 * Executes the remote job.
-	 *
-	 * @param streamGraph
-	 *            Stream Graph to execute
-	 * @param jarFiles
-	 * 			  List of jar file URLs to ship to the cluster
-	 * @return The result of the job execution, containing elapsed time and accumulators.
-	 */
-	@Override
-	protected JobExecutionResult executeRemotely(StreamGraph streamGraph, List<URL> jarFiles) throws ProgramInvocationException {
+	protected List<URL> getJarFiles() throws ProgramInvocationException {
 		URL jarUrl;
 		try {
 			jarUrl = flinkILoop.writeFilesToDisk().getAbsoluteFile().toURI().toURL();
 		} catch (MalformedURLException e) {
+			// TODO: we don't have the actual jobID at this point
 			throw new ProgramInvocationException("Could not write the user code classes to disk.",
-				streamGraph.getJobGraph().getJobID(), e);
+				new JobID(), e);
```

Comment:
That will generate a random `JobID`, not sure about this. Let's use the constructor without the JobID.